oauth2_provider/views.py

```diff
@@ -58,10 +63,43 @@ class AccessTokenView(provider.oauth2.views.AccessTokenView):
     and available to the claim handlers configured by `OAUTH_OIDC_ID_TOKEN_HANDLERS`
 
     """
+    grant_types = provider.oauth2.views.AccessTokenView.grant_types + ["edx_session"]
 
     # Add custom authentication provider, to support email as username.
     authentication = (provider.oauth2.views.AccessTokenView.authentication +
-                      (PublicPasswordBackend, ))
+                      (PublicPasswordBackend, ExistingSessionBackend))
+
+    def edx_session(self, request, data, client):
+        """
+        Handle ``grant_type=edx_session`` requests.
+        """
+        grant_data = self.get_edx_session_grant(request, request.POST, client)
+        at = self.create_access_token(request, grant_data['user'], grant_data['scope'], client)
```

Comment:
@rlucioni this is somewhat of a shortcut / implementation detail.  in order to generate an id_token (which is the JWT we want to respond with), the oauth2 library wants an access token to exist, and uses that to build the id_token.  The HTTP response contains both.

Longer term we want the access_token itself to be the same as the id_token, a change which should be opaque to and compatible with existing clients.  But I'm trying to bootstrap things with the smallest number of changes first.